rust/src/utils/sql.rs

```diff
@@ -16,18 +16,41 @@
 
 use sqlparser::{
     ast::{Expr, SetExpr, Statement},
-    dialect::GenericDialect,
+    dialect::{Dialect, GenericDialect},
     parser::Parser,
     tokenizer::{Token, Tokenizer},
 };
 
 use crate::{Error, Result};
 
+#[derive(Debug, Default)]
+struct LanceDialect(GenericDialect);
+
+impl LanceDialect {
+    fn new() -> Self {
+        Self(GenericDialect {})
+    }
+}
+
+impl Dialect for LanceDialect {
```

Comment:
Given that we are going to build our own dialect, i am wondering whether we should put 

https://github.com/eto-ai/sqlparser-rs/tree/lei/double_eq back to `lance` as well, so that we can keep track with the upstream sqlparser